src/components/wasteCalendar/WasteReminderSettings.tsx

```diff
@@ -0,0 +1,361 @@
+
+  let minutesString = minutes.toString();
+  if (minutes < 10) minutesString = `0${minutesString}`;
+
+  return `${hours}:${minutesString} Uhr`;
+};
+
+const keyExtractor = (item: string) => item;
+
+const CategoryEntry = ({
+  active,
+  categoryKey,
+  categoryName,
+  dispatch
+}: {
+  categoryKey: string;
+  categoryName: string;
+  active: boolean;
+  dispatch: React.Dispatch<ReminderSettingsAction>;
+}) => {
+  const [switchValue, setSwitchValue] = useState(active);
+
+  const toggleSwitch = useCallback(
+    (value) => {
+      setSwitchValue(value);
+      dispatch({
+        type: ReminderSettingsActionType.UPDATE_ACTIVE_TYPE,
+        payload: { key: categoryKey, value }
+      });
+    },
+    [categoryKey, dispatch, setSwitchValue]
+  );
+
+  const onPress = useCallback(() => {
+    toggleSwitch(!switchValue);
+  }, [switchValue, toggleSwitch]);
+
+  useEffect(() => {
+    setSwitchValue(active);
+  }, [active, setSwitchValue]);
+
+  return (
+    <ListItem
+      title={<RegularText>{categoryName}</RegularText>}
+      bottomDivider
+      rightIcon={<Switch switchValue={switchValue ?? false} toggleSwitch={toggleSwitch} />}
+      onPress={onPress}
+      delayPressIn={0}
+      Component={Touchable}
+      accessibilityLabel={`${categoryName} (Taste)`}
+    />
+  );
+};
+
+export const WasteReminderSettings = ({
+  types,
+  locationData
+}: {
+  types: WasteTypeData;
+  locationData: { city: string; street: string; zip: string };
+}) => {
+  const [state, dispatch] = useReducer(reminderSettingsReducer, initialSettings);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [showDatePicker, setShowDatePicker] = useState(false);
+
+  const streetName = locationData.street;
+
+  const { onDayBefore, reminderTime } = state;
+
+  // this state stores the value of the DateTimePicker
+  const [localSelectedTime, setLocalSelectedTime] = useState<Date>(new Date());
+
+  const loadStoredSettings = useCallback(
+    async (street: string) => {
+      setLoading(true);
+      setError(false);
+      const storedSettings = await getReminderSettings();
+
+      if (!areValidReminderSettings(storedSettings)) {
+        setError(true);
+        setLoading(false);
+      } else {
+        const newSettings: ReminderSettings = parseReminderSettings(storedSettings, street);
+
+        // TODO: fix label for stored settings of inactive types ans show them as well
+        // initialize uninitialized types with false
+        Object.keys(types).forEach((key) => {
+          newSettings.activeTypes[key] ??= { active: false };
+        });
+
+        dispatch({ type: ReminderSettingsActionType.OVERWRITE, payload: newSettings });
+        setLocalSelectedTime(newSettings.reminderTime);
+
+        setLoading(false);
+      }
+    },
+    [dispatch, setError, setLoading, types]
+  );
+
+  const onAbortIOS = useCallback(() => {
+    setShowDatePicker(false);
+    setLocalSelectedTime(state.reminderTime);
```

Comment:
this resets the selected time, right? because with `onDatePickerChange` the `newTime` is already set per `setLocalSelectedTime` and when cancelling the picker/modal, the previous value needs to be stored?